diff -durN Osmose-0-9-1.orig/Makefile Osmose-0-9-1/Makefile
--- Osmose-0-9-1.orig/Makefile	2009-11-01 11:06:49.000000000 +0100
+++ Osmose-0-9-1/Makefile	2009-11-07 11:47:39.000000000 +0100
@@ -42,7 +42,7 @@
 # Compiler/Linker Flags
 # add -O3 to CFLAGS for speedup -pg on LFLAGS and CFLAGS for gprof
 CFLAGS =  -Wall -D__USE_UNIX98 -O3
-LFLAGS =  -lSDL -lGL -lz
+LFLAGS =  -lSDL -lGL -lz -lminizip 
 AFLAGS =  -f elf
 CC  = g++
 
@@ -50,10 +50,10 @@
 .cpp.o:
 	$(CC) $(CFLAGS) -c $*.cpp -o $*.o
 
-all:  libs disasm osmose stripexe
+all:  disasm osmose
 
 osmose: $(z80_lib) $(OSM_OBJS) $(OPT_OBJS)
-	$(CC)  $(CFLAGS) $(OSM_OBJS) $(Z80_LIB) $(OPT_OBJS) $(UNZIP_LIB) $(Z_LIB) -o $(OSM_EXE) $(LFLAGS)
+	$(CC)  $(CFLAGS) $(OSM_OBJS) $(Z80_LIB) $(OPT_OBJS) -o $(OSM_EXE) $(LFLAGS)
 
 disasm: $(DISASM_OBJS)
 
diff -durN Osmose-0-9-1.orig/MemoryMapper.h Osmose-0-9-1/MemoryMapper.h
--- Osmose-0-9-1.orig/MemoryMapper.h	2009-10-25 15:54:31.000000000 +0100
+++ Osmose-0-9-1/MemoryMapper.h	2009-11-07 11:44:22.000000000 +0100
@@ -17,7 +17,7 @@
 #include <iostream>
 #include <fstream>
 #include <iomanip>
-#include "unzip/unzip.h"
+#include <minizip/unzip.h>
 #include "Options.h"
 #include "Definitions.h"
 #include "DebugEventThrower.h"
