diff -durN Osmose.orig/Makefile Osmose/Makefile
--- Osmose.orig/Makefile	2006-11-29 16:18:08.000000000 +0100
+++ Osmose/Makefile	2008-01-19 14:14:44.000000000 +0100
@@ -1,11 +1,3 @@
-#Zlib
-Z_LIB_DIR=  ./zlib/
-Z_LIB    =  $(Z_LIB_DIR)libz.a
-
-#unzip based on zlib.
-UNZIP_DIR=  ./unzip/
-UNZIP_LIB=  $(UNZIP_DIR)unzip.a
-
 DISASM_DIR = ./z80_disasm
 
 # Uncomment these lines before builting with internal debugger.
@@ -54,7 +46,7 @@
 # Compiler/Linker Flags
 # add -O3 to CFLAGS for speedup -pg on LFLAGS and CFLAGS for gprof
 CFLAGS =  -Wall -D__USE_UNIX98 -O3
-LFLAGS =  -lSDL
+LFLAGS =  -lz -lminizip -lSDL
 AFLAGS =  -f elf
 CC  = g++
 C_COMP = gcc
@@ -64,10 +56,10 @@
 .cpp.o:
 	$(CC) $(CFLAGS) -c $*.cpp -o $*.o
 
-all:  libs disasm osmose stripexe
+all:  disasm osmose
 
 osmose: $(z80_lib) $(OSM_OBJS) $(OPT_OBJS)
-	$(CC)  $(CFLAGS) $(OSM_OBJS) $(Z80_LIB) $(OPT_OBJS) $(UNZIP_LIB) $(Z_LIB) -o $(OSM_EXE) $(LFLAGS)
+	$(CC)  $(CFLAGS) $(OSM_OBJS) $(Z80_LIB) $(OPT_OBJS) -o $(OSM_EXE) $(LFLAGS)
 
 disasm: $(DISASM_OBJS)
 
@@ -78,19 +70,12 @@
 	rm -f $(DISASM_DIR)/*.o
 	rm -f ./cpu/*.o
 	rm -f *.o core $(OSM_EXE)
-	rm -f $(Z_LIB_DIR)*.o $(Z_LIB_DIR)*.a
-	rm -f $(UNZIP_DIR)*.o $(UNZIP_DIR)*.a
 
-# Clear all objects except ./cpu and zlib.
+# Clear all objects except ./cpu.
 # This is done because compilation of CPU is long.
 miniclean:
 	rm -f *.o core $(OSM_EXE)
 
-libs:
-	cd $(Z_LIB_DIR) && make libz.a
-	cd $(UNZIP_DIR) && make unzip.a
-
-
 
 
 
diff -durN Osmose.orig/MemoryMapper.h Osmose/MemoryMapper.h
--- Osmose.orig/MemoryMapper.h	2006-11-27 18:52:42.000000000 +0100
+++ Osmose/MemoryMapper.h	2008-01-19 12:21:57.000000000 +0100
@@ -17,7 +17,7 @@
 #include <iostream>
 #include <fstream>
 #include <iomanip>
-#include "unzip/unzip.h"
+#include <minizip/unzip.h>
 #include "Options.h"
 #include "Definitions.h"
 #include "DebugEventThrower.h"
