diff -durN Osmose-0-9-2.orig/Makefile Osmose-0-9-2/Makefile
--- Osmose-0-9-2.orig/Makefile	2009-12-10 19:33:32.000000000 +0100
+++ Osmose-0-9-2/Makefile	2009-12-12 12:58:46.652385568 +0100
@@ -43,7 +43,7 @@
 # add -O3 to CFLAGS for speedup -pg on LFLAGS and CFLAGS for gprof
 # remove -DUSE_ISO_IEC_6429 to use a monochrom terminal.
 CFLAGS =  -Wall -D__USE_UNIX98 -DUSE_ISO_IEC_6429 -O3
-LFLAGS =  -lSDL -lGL -lz
+LFLAGS =  -lSDL -lGL -lz -lminizip
 AFLAGS =  -f elf
 CC  = g++
 
@@ -51,10 +51,10 @@
 .cpp.o:
 	$(CC) $(CFLAGS) -c $*.cpp -o $*.o
 
-all:  libs disasm osmose stripexe
+all:  disasm osmose
 
 osmose: $(z80_lib) $(OSM_OBJS) $(OPT_OBJS)
-	$(CC)  $(CFLAGS) $(OSM_OBJS) $(Z80_LIB) $(OPT_OBJS) $(UNZIP_LIB) $(Z_LIB) -o $(OSM_EXE) $(LFLAGS)
+	$(CC)  $(CFLAGS) $(OSM_OBJS) $(Z80_LIB) $(OPT_OBJS) -o $(OSM_EXE) $(LFLAGS)
 
 disasm: $(DISASM_OBJS)
 
diff -durN Osmose-0-9-2.orig/MemoryMapper.h Osmose-0-9-2/MemoryMapper.h
--- Osmose-0-9-2.orig/MemoryMapper.h	2009-12-10 19:35:31.000000000 +0100
+++ Osmose-0-9-2/MemoryMapper.h	2009-12-12 12:59:02.368385484 +0100
@@ -17,7 +17,7 @@
 #include <iostream>
 #include <fstream>
 #include <iomanip>
-#include "unzip/unzip.h"
+#include <minizip/unzip.h>
 #include "Options.h"
 #include "Definitions.h"
 #include "DebugEventThrower.h"
