--- Osmose/MemoryMapper.cpp.orig	2006-11-27 17:56:30.000000000 +0000
+++ Osmose/MemoryMapper.cpp	2008-02-01 21:31:34.000000000 +0000
@@ -16,6 +16,8 @@
 #include "Options.h"
 #include "RomSpecificOption.h"
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 
 extern Options opt;
 
--- Osmose/Options.cpp.orig	2006-11-24 10:35:30.000000000 +0000
+++ Osmose/Options.cpp	2008-02-01 21:30:20.000000000 +0000
@@ -14,6 +14,7 @@
 *****************************************************************************/
 #include "Options.h"
 #include "MemoryMapper.h"
+#include <string.h>
 
 void Options::reset()
 {
--- Osmose/WaveWriter.cpp.orig	2006-11-06 11:08:38.000000000 +0000
+++ Osmose/WaveWriter.cpp	2008-02-01 21:42:32.000000000 +0000
@@ -13,6 +13,7 @@
 * URL: http://bcz.emu-france.com/
 *****************************************************************************/
 #include "WaveWriter.h"
+#include <stdlib.h>
 
 /*----------------------------------------------------------------------*/
 /* Constructor: Initialise structures ChunkFormat, Data Format, and	*/
